rust/db_handling/src/interface_signer.rs

```diff
@@ -411,26 +430,45 @@ pub fn derive_prep(
     network_specs_key: &NetworkSpecsKey,
     collision: Option<(MultiSigner, AddressDetails)>,
     suggest: &str,
-) -> Result<String, ErrorSigner> {
+    keyboard: bool,
+) -> Result<MDeriveKey, ErrorSigner> {
     let network_specs = get_network_specs(database_name, network_specs_key)?;
-    match collision {
-        Some((multisigner, address_details)) => {
-            let address_base58 = print_multisigner_as_base58(&multisigner, Some(network_specs.base58prefix));
-            let hex_identicon = hex::encode(make_identicon_from_multisigner(&multisigner));
-            let collision_display = format!("\"base58\":\"{}\",\"path\":\"{}\",\"has_pwd\":{},\"identicon\":\"{}\",\"seed_name\":\"{}\"", address_base58, address_details.path, address_details.has_pwd, hex_identicon, seed_name);
-            Ok(format!("\"seed_name\":\"{}\",\"network_title\":\"{}\",\"network_logo\":\"{}\",\"network_specs_key\":\"{}\",\"suggested_derivation\":\"{}\",\"collision\":{{{}}}", seed_name, network_specs.title, network_specs.logo, hex::encode(network_specs_key.key()), suggest, collision_display))
-        },
-        None => Ok(format!("\"seed_name\":\"{}\",\"network_title\":\"{}\",\"network_logo\":\"{}\",\"network_specs_key\":\"{}\",\"suggested_derivation\":\"{}\"", seed_name, network_specs.title, network_specs.logo, hex::encode(network_specs_key.key()), suggest)),
-    }
+    let derivation_check = collision.map(|(multisigner, address_details)| {
```

Comment:
derivation_check field should not be an option: it should always be present and explicitly tell UI, whether button is pressable and if so what should it do; even when there are no collisions, it should be there. Right now this behavior results in following bug: 

1. create seed without seed key 
2. try to derive key - button is grey (incorrect)
3. type a character and remove it - button is blue (correct)

This should behave exactly as `db_handling::interface_signer::dynamic_path_check` derivation_check constructor